scripts/travis/save-logs.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import sys
+import subprocess
+import mmap
+import os
+
+from datetime import datetime, timedelta
+
+def tail(filename, n):
```

Comment:
Yes ,we can call "tail" shell. but I do not think that make any difference.

Code Sample:
```
#!/usr/bin/python
# -*- coding: utf-8 -*-

import subprocess 
filename = '/tmp/result.log'  
command='tail -1000f '+filename
popen=subprocess.Popen(command,stdout=subprocess.PIPE,stderr=subprocess.PIPE,shell=True)  
while True:  
    line=popen.stdout.readline().strip()  
	print line
```